bpf/lib/nat.h

```diff
@@ -871,6 +958,66 @@ snat_v4_rev_nat(struct __ctx_buff *ctx, const struct ipv4_nat_target *target)
 			tuple.dport = icmphdr.un.echo.id;
 			tuple.sport = 0;
 			break;
+		case ICMP_DEST_UNREACH:
+			switch (icmphdr.code) {
+			case ICMP_FRAG_NEEDED: {
+				struct iphdr iphdr;
+				__be16 identifier;
+				__u8 type;
+				__u32 icmpoff = off + sizeof(icmphdr);
+
+				/* According to the RFC 5508, any networking
+				 * equipment that is responding with an ICMP Error
+				 * packet should embed the original packet in its
+				 * response.
+				 */
+
+				if (ctx_load_bytes(ctx, icmpoff, &iphdr,
+						   sizeof(iphdr)) < 0)
+					return DROP_INVALID;
+
+				/* From the embedded IP headers we should be able
+				 * to determine corresponding protocol, IP src/dst
+				 * of the packet sent to resolve the NAT session.
+				 */
+				tuple.nexthdr = iphdr.protocol;
+				tuple.saddr = iphdr.daddr;
+				tuple.daddr = iphdr.saddr;
+
+				icmpoff += ipv4_hdrlen(&iphdr);
+				switch (tuple.nexthdr) {
```

Comment:
If you don't mind, I have opened case #23484, could you assign myself on it? I will work on this missing piece of support but would like if possible not mix more change on this one that to avoid any unpredictable yet blockers, sounds reasonable?

Reply:
Have a look how `snat_v4_rev_nat()` does the port extraction, and `snat_v4_rewrite_ingress()` handles the csum update. Imho it's trivial, and good practice to keep the code base in-sync. It's all hidden behind `ENABLE_SCTP`, so shouldn't impact complexity by default (if that's your concern).

Reply:
Looks good, ty!